oas.go

```diff
@@ -31,12 +36,30 @@ type ServerObject struct {
 }
 
 type InfoObject struct {
-	Title          string         `json:"title"`
-	Description    string         `json:"description,omitempty"`
-	TermsOfService string         `json:"termsOfService,omitempty"`
-	Contact        *ContactObject `json:"contact,omitempty"`
-	License        *LicenseObject `json:"license,omitempty"`
-	Version        string         `json:"version"`
+	Title          string          `json:"title"`
+	Description    *ReffableString `json:"description,omitempty"`
+	TermsOfService string          `json:"termsOfService,omitempty"`
+	Contact        *ContactObject  `json:"contact,omitempty"`
+	License        *LicenseObject  `json:"license,omitempty"`
+	Version        string          `json:"version"`
+}
+
+// Wrapper for a string that may be of the form `$ref:foo`
+// denoting it should be json encoded as `{ "$ref": "foo" }`
+type ReffableString struct {
+	Value string
+}
+
+type reference struct {
+	Ref string `json:"$ref"`
+}
+
+func (r ReffableString) MarshalJSON() ([]byte, error) {
+	if strings.HasPrefix(r.Value, "$ref:") {
```

Comment:
Goas should error if `r.Value` contains _exactly_ the string `"$ref:"`. Right now I believe we'll marshall effectively an empty ref, which may lead to undefined behavior when we try to parse and use the spec.

Let's add a test for this scenario, too.

Reply:
Good catch! See blow, plus tests! 🎸